pallets/connectors/src/lib.rs

```diff
@@ -453,11 +514,16 @@ pub mod pallet {
 				Error::<T>::UnauthorizedTransfer
 			);
 
-			ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+			// Ensure pool and tranche exist and derive invest id
+			let invest_id = Self::derive_invest_id(pool_id, tranche_id)?;
+			ensure!(
+				CurrencyIdOf::<T>::is_tranche_token(invest_id.clone().into()),
+				Error::<T>::InvalidTransferCurrency
+			);
```

Comment:
The `TrancheCurrency::generate` will always generate the right currency. So this check is not needed IMO.